lib/widgets/compose_box.dart

```diff
@@ -1675,12 +1825,146 @@ class ComposeBox extends StatefulWidget {
 /// The interface for the state of a [ComposeBox].
 abstract class ComposeBoxState extends State<ComposeBox> {
   ComposeBoxController get controller;
+
+  /// Switch the compose box to editing mode.
+  ///
+  /// If there is already text in the compose box, gives a confirmation dialog
+  /// to confirm that it is OK to discard that text.
+  ///
+  /// If called from the message action sheet, fetches the raw message content
+  /// to fill in the edit-message compose box.
+  ///
+  /// If called by tapping a message in the message list with 'EDIT NOT SAVED',
+  /// fills the edit-message compose box with the content the user wanted
+  /// in the edit request that failed.
+  void startEditInteraction(int messageId);
+
+  /// Switch the compose box back to regular non-edit mode, with no content.
+  void endEditInteraction();
 }
 
 class _ComposeBoxState extends State<ComposeBox> with PerAccountStoreAwareStateMixin<ComposeBox> implements ComposeBoxState {
   @override ComposeBoxController get controller => _controller!;
   ComposeBoxController? _controller;
 
+  @override
+  void startEditInteraction(int messageId) async {
+    if (await _abortBecauseContentInputNotEmpty()) return;
+    if (!mounted) return;
+
+    final store = PerAccountStoreWidget.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+
+    switch (store.getEditMessageErrorStatus(messageId)) {
+      case null:
+        _editFromRawContentFetch(messageId);
```

Comment:
bump https://github.com/zulip/zulip-flutter/pull/1498#discussion_r2092232833: put the two non-error cases next to each other

Reply:
Ah! I put `null` and `false` together as the "two non-error cases", thinking of "error" as it's used in `getEditMessageErrorStatus` which produces this value:

```dart
  @override
  bool? getEditMessageErrorStatus(int messageId) =>
    _editMessageRequests[messageId]?.hasError;
```

But I see now what you mean, that the error case is the one where we show an error dialog. 😛